### PR TITLE
Work on find all references / nameres fixes

### PR DESCRIPTION
**Pull request by Roldak**

### CODE REVIEW

File: ada/language/ast.py [hunk 1]
```diff
@@ -5860,7 +5860,7 @@ def subprogram_symbol():
 
     @langkit_property(public=True, return_type=T.DefiningName.entity)
     def xref():
-        return Entity.parent.referenced_decl.defining_name
+        return Entity.parent.referenced_decl._.defining_name
```
Comment (raph-amiard) [inline ada/language/ast.py]: Which crash/where is the test ?

File: ada/testsuite/tests/properties/find_all_references/test.out [hunk 2]
```diff
@@ -1,29 +1,36 @@
 All references to A from a.ads:
```
Comment (raph-amiard) [inline ada/testsuite/tests/properties/find_all_references/test.out]: You don't seem to test find_all_refs on a built-in operator right ?

File: ada/language/ast.py [hunk 3]
```diff
@@ -8142,7 +8142,20 @@ def dottable_subp_of():
                 # However, since private types don't have components, this
                 # should not ever be a problem with legal Ada.
                 (t.is_tagged_type | t.private_completion._.is_tagged_type)
-                & (t.declarative_scope == bd.declarative_scope),
+                & bd.declarative_scope.then(lambda ds: Or(
```
Comment (raph-amiard) [inline ada/language/ast.py]: Plz comment dat stuff. Also plz add the test.

File: ada/testsuite/tests/properties/find_all_references/test.out (hunk 2 above)
Comment (Roldak) [reply ada/testsuite/tests/properties/find_all_references/test.out]: U fuh'in w0t m8??

**Pull request merged**